internal/firecracker/server.go

```diff
@@ -136,6 +137,7 @@ func StartFirecrackerServer(
 	if err := cmd.Start(); err != nil {
 		panic(errors.Join(ErrCouldNotStartFirecrackerServer, err))
 	}
+	server.VMPid = cmd.Process.Pid
```

Comment:
We'll need to check if `Process != nil` here IIRC. It's unlikely but possible to kill the process in between those two step, and since it _can_ be `nil` (see the check in `Close()`), so if it is we'll need to return an error

Reply:
Nvm, the reason for the `nil` check was actually because we `defer Close()` even if we return earlier than `cmd.Start()` :)

Reply:
(See https://pkg.go.dev/os/exec#Cmd.Start)